core-bundle/tests/Translation/TranslatorTest.php

```diff
@@ -220,6 +220,53 @@ public function testUsesDecoratedCatalogues(): void
         $this->assertInstanceOf(MessageCatalogue::class, $catalogues[1]);
     }
 
+    public function testRestoresPreviousTranslationsInGlobals(): void
+    {
+        $originalTranslator = $this->createMock(BaseTranslator::class);
+        $originalTranslator
+            ->expects($this->never())
+            ->method('trans')
+        ;
+
+        $originalTranslator
+            ->expects($this->exactly(2))
+            ->method('getLocale')
+            ->willReturn('en')
+        ;
+
+        $originalTranslator
+            ->method('getCatalogue')
+            ->willReturnCallback(
+                function ($locale) {
+                    $catalogue = $this->createMock(MessageCatalogueInterface::class);
+                    $catalogue
+                        ->method('getLocale')
+                        ->willReturn($locale ?? 'de')
```

Comment:
```suggestion
                        ->willReturn($locale ?? 'en')
```
shouldn’t this fall back to the default?